src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -55,8 +55,7 @@ struct SpectralTransform{
     
     # SCRATCH MEMORY FOR FOURIER NOT YET LEGENDRE TRANSFORMED AND VICE VERSA
     # state is undetermined, only read after writing to it
-    scratch_memory_north::ArrayComplexType
-    scratch_memory_south::ArrayComplexType
+    scratch_memory::ScratchMemory{NF, ArrayComplexType} 
     scratch_memory_grid::VectorType                 # scratch memory with 1-stride for FFT output
     scratch_memory_spec::VectorComplexType
     scratch_memory_column_north::VectorComplexType  # scratch memory for vertically batched Legendre transform
```

Comment:
But these should be part of the scratch memory too no? Like they are actually used, though much smaller, I agree.

Reply:
Yeah, as I wrote above, I had a version where I had those in as well first. Strictly speaking it's not necessary for Enzyme's purposes. Would also require (minor) changes to `_fourier!` and `_legendre!`. I am happy to look into it again though, I also find it a bit inconsistent like that.